Automattic/newspack-plugin
# Unified Ad Wizard

### All Submissions:






### Changes proposed in this Pull Request:

This PR is a complete refactor of the Ad wizards, based on a variety of new directions in UI design and code organization.

<details>
1. Main screen. Lists all currently supported Ad Services. Enabling AdSense or Ad Manager simply sets an option. Enabling WordAds activates the Jetpack module. If Jetpack plan is too low, a notification will be shown.

<img width="1668" alt="Screen Shot 2019-07-24 at 9 33 35 AM" src="https://user-images.githubusercontent.com/1477002/61798109-bb73cf80-adf6-11e9-8645-1ba5a697727c.png">

2. Global Settings. This is where predefined slots can be filled. Note that there is no implementation of this in the theme yet, so for now these settings will not affect the published site in any way. For each placement, once enable there will be a dropdown allowing selection of the enabled services. If Google Ad Manager is selected a second Select appears containing all defined Ad Units. The WordAds configure link also goes to this screen, since this is the only WordAds configuration necessary. 

<img width="1668" alt="Screen Shot 2019-07-24 at 9 34 29 AM" src="https://user-images.githubusercontent.com/1477002/61798407-3e952580-adf7-11e9-8fe8-934412b6aca6.png">

3. AdSense setup. This is a `Handoff` to Site Kit, pretty much unchanged from the AdSense wizard currently on `master`..

<img width="1668" alt="Screen Shot 2019-07-24 at 9 34 43 AM" src="https://user-images.githubusercontent.com/1477002/61798580-86b44800-adf7-11e9-84e6-3fa0405418ed.png">

4. Ad Manager Ad Units. This is where ad units are listed, edited, deleted, and created. More or less unchanged from the current Google Ad Manager wizard on `master`.

<img width="1668" alt="Screen Shot 2019-07-24 at 9 34 51 AM" src="https://user-images.githubusercontent.com/1477002/61798661-b06d6f00-adf7-11e9-9897-c6730b4cdd0b.png">

5. Ad Unit Detail. Create or edit Google Ad Manager Ad Units here. This UI is largely unchanged from the Ad Manager wizard on `master.` There is now support for AMP/non-AMP codes. Also, the name of the ad service is saved with each unit (`google_ad_manager`), which means this can easily be extended to support other similar ad services.

<img width="1668" alt="Screen Shot 2019-07-24 at 9 34 55 AM" src="https://user-images.githubusercontent.com/1477002/61798842-08a47100-adf8-11e9-98ab-093d2b538cf3.png">

6. Global code. Google Ad Manager header code. 

<img width="1668" alt="Screen Shot 2019-07-24 at 9 34 58 AM" src="https://user-images.githubusercontent.com/1477002/61798893-21148b80-adf8-11e9-8cce-ee6800cee408.png">

</details>

### How to test the changes in this Pull Request:

1. `npm ci && npm run clean && npm run build:webpack`
2. Newspack Ads will need to be cloned into `wp-content/themes`, checked out to `update/extend-model`, and build (`npm ci && npm run build:webpack`)
3. The screens are described in detail above, so it's probably best to play around with them and see what comes up.
4. Use the sample ad codes at p2-pamTN9-kv to test

### Other information:

This includes two open PRs which will need to ship before this can: 
- https://github.com/Automattic/newspack-plugin/pull/203
- https://github.com/Automattic/newspack-plugin/pull/202

As well as https://github.com/Automattic/newspack-ads/pull/6
